src/bin/common/solving_loop.ml

```diff
@@ -96,14 +105,16 @@ let main () =
           state.solver_ctx.ctx
         in
         let cnf = List.rev @@ Cnf.make l td in
-        let () = solve all_context (cnf, name) in
+        let sat_env = solve state.solver_ctx.sat_env all_context (cnf, name) in
```

Comment:
It is almost correct. The field `sat_env` in the state contents the last environment produced by the function solve. It's not used by the next call to the SAT solver. But I should replace the field for the sat environment by an option since the SAT solver can timeout. In this case, the current code keeps the last produced model, which is counterintuitive.

Reply:
ah, right, `solve` takes the `sat_env` just to return it in case of timeout -- indeed using an option is better

I suggest using a separate state key for the sat env in dolmen and simply ignoring the new result of `solve` (`let _ = solve ...`) in the `typed_loop` since AFAIK the legacy frontend has no way to display the models anyways.

Reply:
Ok, I will ignore the result in the legacy frontend.